openrtb_ext/device.go

```diff
@@ -12,14 +14,58 @@ const PrebidExtKey = "prebid"
 
 // ExtDevice defines the contract for bidrequest.device.ext
 type ExtDevice struct {
+	// Attribute:
+	//   atts
+	// Type:
+	//   integer; optional - iOS Only
+	// Description:
+	//   iOS app tracking authorization status.
+	// Extension Spec:
+	//   https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/extensions/community_extensions/skadnetwork.md
+	ATTS *IOSAppTrackingStatus `json:"atts"`
+
+	// Attribute:
+	//   prebid
+	// Type:
+	//   object; optional
+	// Description:
+	//   Prebid extensions for the Device object.
 	Prebid ExtDevicePrebid `json:"prebid"`
 }
 
-// Pointer to interstitial so we do not force it to exist
+// IOSAppTrackingStatus describes the values for iOS app tracking authorization status.
+type IOSAppTrackingStatus int
+
+// Values of the IOSAppTrackingStatus enumeration.
+const (
+	IOSAppTrackingStatusNotDetermined IOSAppTrackingStatus = 0
+	IOSAppTrackingStatusRestricted    IOSAppTrackingStatus = 1
+	IOSAppTrackingStatusDenied        IOSAppTrackingStatus = 2
+	IOSAppTrackingStatusAuthorized    IOSAppTrackingStatus = 3
+)
+
+// IsKnownIOSAppTrackingStatus returns true if the value is a known iOS app tracking authorization status.
+func IsKnownIOSAppTrackingStatus(v int64) bool {
+	switch IOSAppTrackingStatus(v) {
+	case IOSAppTrackingStatusNotDetermined:
+		return true
+	case IOSAppTrackingStatusRestricted:
+		return true
+	case IOSAppTrackingStatusDenied:
+		return true
+	case IOSAppTrackingStatusAuthorized:
+		return true
+	default:
+		return false
```

Comment:
Since `IOSAppTrackingStatusNotDetermined`, `IOSAppTrackingStatusRestricted`, `IOSAppTrackingStatusDenied`, and `IOSAppTrackingStatusAuthorized` are equal to ints from 0 to 3, `IsKnownIOSAppTrackingStatus` could probably be simplified to:
```
48   func IsKnownIOSAppTrackingStatus(v int64) bool {
49 -     switch IOSAppTrackingStatus(v) {
50 -     case IOSAppTrackingStatusNotDetermined:
51 -         return true
52 -     case IOSAppTrackingStatusRestricted:
53 -         return true
54 -     case IOSAppTrackingStatusDenied:
55 -         return true
56 -     case IOSAppTrackingStatusAuthorized:
57 -         return true
58 -     default:
59 -         return false
60 -     }
   +     if v < IOSAppTrackingStatusNotDetermined || v > IOSAppTrackingStatusAuthorized {
   +         return false
   +     }
   +     return true
61   }
openrtb_ext/device.go
```

Reply:
If it's alright with you, I'd prefer to keep it as a switch. The fact that the enumeration values are sequential are not a guaranteed feature of the value. I don't believe this would have a performance impact.